testing/trino-tests/src/test/java/io/trino/execution/resourcegroups/db/TestQueuesDb.java

```diff
@@ -369,6 +371,29 @@ public void testNonLeafGroup()
         assertEquals(queryRunner.getCoordinator().getDispatchManager().getQueryInfo(invalidResourceGroupQuery).getErrorCode(), INVALID_RESOURCE_GROUP.toErrorCode());
     }
 
+    @Test
+    public void testUpdateSoftMemoryLimit()
+            throws Exception
+    {
+        // trigger resource group creation
+        queryRunner.execute("SELECT COUNT(*), clerk FROM orders GROUP BY clerk");
+        InternalResourceGroupManager<?> manager = queryRunner.getCoordinator().getResourceGroupManager().get();
+        DbResourceGroupConfigurationManager dbConfigurationManager = (DbResourceGroupConfigurationManager) manager.getConfigurationManager();
+
+        dao.updateResourceGroup(2, "bi-${USER}", "1MB", 3, 2, 2, null, null, null, null, null, 1L, TEST_ENVIRONMENT);
+        dbConfigurationManager.load();
+        dao.updateResourceGroup(2, "bi-${USER}", "20%", 3, 2, 2, null, null, null, null, null, 1L, TEST_ENVIRONMENT);
+        dbConfigurationManager.load();
+        dao.updateResourceGroup(2, "bi-${USER}", "123MB", 3, 2, 2, null, null, null, null, null, 1L, TEST_ENVIRONMENT);
+        dbConfigurationManager.load();
+
+        // wait for SqlQueryManager which enforce memory limits per second
+        MILLISECONDS.sleep(2000);
+        ResourceGroupInfo groupInfo = manager.tryGetResourceGroupInfo(new ResourceGroupId(new ResourceGroupId("global"), "bi-user"))
+                .orElseThrow(() -> new IllegalStateException("Resource group not found"));
+        assertEquals(groupInfo.getSoftMemoryLimit(), DataSize.of(123, MEGABYTE));
```

Comment:
we should also assert that the initial change to percent based value also took effect using an assert similar to this.

Reply:
> we should also assert that the initial change to percent based value also took effect using an assert similar to this.

Not sure if there is any better way to do assertion, but I found this might cause flaky tests. (the Math.round part)
```java
        ResourceGroupInfo groupInfo =
                manager.tryGetResourceGroupInfo(new ResourceGroupId(new ResourceGroupId("global"), "bi-user"))
                        .orElseThrow(() -> new IllegalStateException("Resource group not found"));
        assertEquals(
                groupInfo.getSoftMemoryLimit(),
                DataSize.of(Math.round(queryRunner.getCoordinator().getClusterMemoryManager().getClusterMemoryBytes() * 0.2), BYTE));
```

Reply:
Thanks for the tips !